src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -72,14 +102,58 @@ private SecretClient GetSecretClient(Uri secretUri)
                 return client;
             }
 
-            if (_credential == null)
+            if (_keyVaultOptions.Credential == null)
             {
                 return null;
             }
 
-            client = new SecretClient(new Uri(secretUri.GetLeftPart(UriPartial.Authority)), _credential);
+            client = new SecretClient(new Uri(secretUri.GetLeftPart(UriPartial.Authority)), _keyVaultOptions.Credential);
             _secretClients.Add(keyVaultId, client);
             return client;
         }
+
+        private void SetSecretInCache(string key, string secretValue)
+        {
+            DateTimeOffset? refreshSecretAt = null;
+
+            if (_keyVaultOptions.SecretRefreshIntervals.TryGetValue(key, out TimeSpan refreshInterval))
+            {
+                refreshSecretAt = DateTimeOffset.UtcNow.Add(refreshInterval);
+            }
+            else if (_keyVaultOptions.DefaultSecretRefreshInterval.HasValue)
+            {
+                refreshSecretAt = DateTimeOffset.UtcNow.Add(_keyVaultOptions.DefaultSecretRefreshInterval.Value);
+            }
+
+            _cachedKeyVaultSecrets[key] = new CachedKeyVaultSecret(secretValue, refreshSecretAt);
+            
+            if (key == _nextRefreshKey)
+            {
+                UpdateNextRefreshableSecretFromCache();
+            }
+            else if ((refreshSecretAt.HasValue && _nextRefreshTime.HasValue && refreshSecretAt.Value < _nextRefreshTime.Value)
+                    || (refreshSecretAt.HasValue && !_nextRefreshTime.HasValue))
+            {
+                _nextRefreshKey = key;
+                _nextRefreshTime = refreshSecretAt.Value;
+            }
+        }
+
+        private void UpdateNextRefreshableSecretFromCache()
+        {
+            _nextRefreshKey = null;
+            _nextRefreshTime = DateTimeOffset.MaxValue;
+
+            foreach (KeyValuePair<string, CachedKeyVaultSecret> secret in _cachedKeyVaultSecrets)
+            {
+                if (secret.Value.RefreshAt.HasValue && secret.Value.RefreshAt.Value < _nextRefreshTime)
+                {
+                    _nextRefreshTime = secret.Value.RefreshAt;
+                    _nextRefreshKey = secret.Key;
+                }
+            }
+
+            _nextRefreshTime = _nextRefreshTime != DateTimeOffset.MaxValue ? _nextRefreshTime : null;
```

Comment:
>_nextRefreshTime = _nextRefreshTime != DateTimeOffset.MaxValue ? _nextRefreshTime : null; [](http://example.com/codeflow?start=12&length=89)

We're doing an assignment, but really it's more like an if statement. Can we write it as an `if` to be more clear.